benches/pk_encryption/ring_lpr.rs

```diff
@@ -0,0 +1,288 @@
+
+            let cipher0 = PolynomialRingZq::from((poly0, &modulus));
+            let cipher1 = PolynomialRingZq::from((poly1, &modulus));
+
+            (cipher0, cipher1)
+        } else {
+            let modulus = qfall_crypto::utils::common_moduli::new_anticyclic(16, 2399).unwrap();
+
+            let poly0: PolyOverZ = serde_json::from_str(SMALL_CIPHER0).unwrap();
+            let poly1: PolyOverZ = serde_json::from_str(SMALL_CIPHER1).unwrap();
+
+            let cipher0 = PolynomialRingZq::from((poly0, &modulus));
+            let cipher1 = PolynomialRingZq::from((poly1, &modulus));
+
+            (cipher0, cipher1)
+        }
+    }
+
+    /// Loads one of the above precomputed variants suitable to:
+    /// - [`SizeN::Small`]: [`RingLPR::default`]
+    /// - [`SizeN::Large`]: [`RingLPR::secure128`]
+    pub(super) fn load_suiting_ring_lpr(size_n: &SizeN) -> RingLPR {
+        if size_n == &SizeN::Large {
+            RingLPR::secure128()
+        } else {
+            RingLPR::default()
+        }
+    }
+}
+
+/// This module contains all functions directly executed in the benchmarking functions.
+pub(super) mod performed_functions {
+    use super::load_precomputed_values::*;
+    use crate::SizeN;
+    use qfall_crypto::construction::pk_encryption::PKEncryptionScheme;
+    use qfall_crypto::construction::pk_encryption::RingLPR;
+    use qfall_math::integer::Z;
+
+    /// Performs a full-cycle of gen, enc, dec with [RingLPR].
+    pub(super) fn ring_lpr_full_cycle_dyn(n: i64) {
+        let msg = Z::ONE;
+        let ring_lpr = RingLPR::new_from_n(n);
+
+        let (pk, sk) = ring_lpr.gen();
+        let cipher = ring_lpr.enc(&pk, &msg);
+        let _ = ring_lpr.dec(&sk, &cipher);
+    }
+
+    /// Performs a cycle of `enc` and `dec` with precomputed [`RingLPR`] parameters and keys and 128-bit security.
+    pub(super) fn ring_lpr_cycle(size_n: &SizeN) {
+        let msg = Z::from(15);
+        let ring_lpr = load_suiting_ring_lpr(size_n);
+
+        let pk = load_precomputed_pk(size_n);
+        let sk = load_precomputed_sk(size_n);
+
+        let cipher = ring_lpr.enc(&pk, &msg);
+        let _ = ring_lpr.dec(&sk, &cipher);
+    }
+
+    /// Performs a cycle of `enc` with precomputed [`RingLPR`] parameters and keys and 128-bit security.
+    pub(super) fn ring_lpr_enc(size_n: &SizeN) {
+        let msg = Z::from(15);
+        let ring_lpr = load_suiting_ring_lpr(size_n);
+
+        let pk = load_precomputed_pk(size_n);
+
+        let _ = ring_lpr.enc(&pk, &msg);
+    }
+
+    /// Performs a cycle of `dec` with precomputed [`RingLPR`] parameters and keys and 128-bit security.
+    pub(super) fn ring_lpr_dec(size_n: &SizeN) {
+        let cipher = load_precomputed_cipher(size_n);
+        let ring_lpr = load_suiting_ring_lpr(size_n);
+
+        let sk = load_precomputed_sk(size_n);
+
+        let _ = ring_lpr.dec(&sk, &cipher);
+    }
+}
+
+/// Benchmark [ring_lpr_full_cycle_dyn] with `n = 8, 16, 32, 64, 128, 256, 512, 1024`
```

Comment:
May want to add them for this and the following docs of the functions
```suggestion
/// Benchmark [`ring_lpr_full_cycle_dyn`] with `n = 8, 16, 32, 64, 128, 256, 512, 1024`
```